amazon-msk/cdk/lib/SecurePublicAccessStack.ts

```diff
@@ -364,6 +385,47 @@ export class SecurePublicAccessStack extends cdk.Stack {
 
     autoScalingGroup.attachToNetworkTargetGroup(targetGroup);
 
+    const metricsNamespace = context.cloudwatch.metrics?.namespace;
+    const interval = context.cloudwatch.metrics?.interval;
+    if (metricsNamespace && interval) {
+
+      const metricWorkerUtilization = new cw.Metric({
+        namespace: metricsNamespace,
+        metricName: 'engine.worker.utilization',
+        statistic: 'Average',
+        period: cdk.Duration.seconds(interval),
+      });
+
+      const metricWorkerCount = new cw.Metric({
+        namespace: metricsNamespace,
+        metricName: 'engine.worker.count',
+        statistic: 'Average',
+        period: cdk.Duration.seconds(interval),
+      });
+
+      const metricOverallWorkerUtilization = new cw.MathExpression({
+        label: 'OverallWorkerUtilization',
+        expression: 'm1 / m2',
+        usingMetrics: {
+          m1: metricWorkerUtilization,
+          m2: metricWorkerCount,
+        },
+      });
+
+      const scalingSteps = context.autoscaling.scalingSteps ??
+        [
+          { upper: 0.30, change: -1 },
+          { lower: 0.80, change: +2 }
+        ]
+
+      autoScalingGroup.scaleOnMetric('WorkerUtilizationStepScaling', {
+        metric: metricOverallWorkerUtilization,
+        adjustmentType: autoscaling.AdjustmentType.CHANGE_IN_CAPACITY,
+        scalingSteps,
+        estimatedInstanceWarmup: cdk.Duration.seconds(context.autoscaling.warmUp),
+        cooldown: cdk.Duration.seconds(context.autoscaling.coolDown)
```

Comment:
```suggestion
        estimatedInstanceWarmup: cdk.Duration.seconds(context.autoscaling.warmup),
        cooldown: cdk.Duration.seconds(context.autoscaling.cooldown)
```

Let's keep the capitalization simple.